src/Components/LifecycleChart/LifecycleChart.tsx

```diff
@@ -368,17 +401,17 @@ const LifecycleChart: React.FC<LifecycleChartProps> = ({
         <ChartGroup horizontal>
           {legendNames.map((s, index) => {
             if (s.datapoints.length === 0) {
-              return;
+              return null;
+            }
+            
+            // Handle hidden series differently - don't include in rendering at all
```

Comment:
This seems a bit suspicious to me, is it really needed? I know the chart is really fragile and super easy to break. 

It seems it happened :( After hiding some lines, the `onHover` is broken (using original version without suggestions). 

https://github.com/user-attachments/assets/ea00b0ae-5395-4440-be36-a4c5409fa847